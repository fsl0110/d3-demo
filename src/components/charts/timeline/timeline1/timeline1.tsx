import React, { PureComponent } from "react";
import { createChart } from "./timeline1Create";

export interface Props {
  data: { artist: string; color: string; startDate: Date; endDate: Date }[];
  width: number;
  height: number;
}

export interface State {
  data: { artist: string; color: string; startDate: Date; endDate: Date }[];
  width: number;
  height: number;
}

export class Timeline1 extends PureComponent<Props, State> {
  readonly state: State = {
    data: [],
    width: 0,
    height: 0
  };

  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
    if (prevState.data !== nextProps.data) {
      return {
        data: nextProps.data
      };
    }
    return null;
  }

  render() {
    const { data, width, height } = this.state;
    return <>{createChart(data, width, height)}</>;
  }
}
