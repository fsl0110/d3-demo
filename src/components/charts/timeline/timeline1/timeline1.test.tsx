import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Timeline1 } from "./timeline1";

it("renders toMatchSnapshot", () => {
  expect(toJson(shallow(<Timeline1 />))).toMatchSnapshot();
});
