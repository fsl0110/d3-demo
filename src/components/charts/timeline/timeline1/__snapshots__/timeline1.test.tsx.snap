// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toMatchSnapshot 1`] = `
<svg
  className="svg__canvas"
  height="300"
  width="600"
/>
`;
